pkg/cliplugins/workspace/plugin/kubeconfig_test.go

```diff
@@ -0,0 +1,653 @@
+)
+
+func TestCreate(t *testing.T) {
+	tests := []struct {
+		name   string
+		config clientcmdapi.Config
+
+		existingWorkspaces []string // existing cluster workspaces
+		markReady          bool
+
+		newWorkspaceName                 string
+		newWorkspaceType                 string
+		useAfterCreation, ignoreExisting bool
+
+		expected *clientcmdapi.Config
+		wantErr  bool
+	}{
+		{
+			name: "happy case, no use",
+			config: clientcmdapi.Config{CurrentContext: "test",
+				Contexts:  map[string]*clientcmdapi.Context{"test": {Cluster: "test", AuthInfo: "test"}},
+				Clusters:  map[string]*clientcmdapi.Cluster{"test": {Server: "https://test/clusters/root:foo"}},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{"test": {Token: "test"}},
+			},
+			newWorkspaceName: "bar",
+		},
+		{
+			name: "create, use after creation, but not ready",
+			config: clientcmdapi.Config{CurrentContext: "test",
+				Contexts:  map[string]*clientcmdapi.Context{"test": {Cluster: "test", AuthInfo: "test"}},
+				Clusters:  map[string]*clientcmdapi.Cluster{"test": {Server: "https://test/clusters/root:foo"}},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{"test": {Token: "test"}},
+			},
+			newWorkspaceName: "bar",
+			useAfterCreation: true,
+			wantErr:          true, // not ready
+		},
+		{
+			name: "create, use after creation",
+			config: clientcmdapi.Config{CurrentContext: "test",
+				Contexts:  map[string]*clientcmdapi.Context{"test": {Cluster: "test", AuthInfo: "test"}},
+				Clusters:  map[string]*clientcmdapi.Cluster{"test": {Server: "https://test/clusters/root:foo"}},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{"test": {Token: "test"}},
+			},
+			newWorkspaceName: "bar",
+			useAfterCreation: true,
+			markReady:        true,
+			expected: &clientcmdapi.Config{CurrentContext: "workspace",
+				Contexts: map[string]*clientcmdapi.Context{
+					"test":                       {Cluster: "test", AuthInfo: "test"},
+					"workspace":                  {Cluster: "workspace", AuthInfo: "test"},
+					"workspace.kcp.dev/previous": {Cluster: "test", AuthInfo: "test"},
+				},
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"test":      {Server: "https://test/clusters/root:foo"},
+					"workspace": {Server: "https://test/clusters/root:foo:bar"},
+				},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{"test": {Token: "test"}},
+			},
+		},
+		{
+			name: "create, already existing",
+			config: clientcmdapi.Config{CurrentContext: "test",
+				Contexts:  map[string]*clientcmdapi.Context{"test": {Cluster: "test", AuthInfo: "test"}},
+				Clusters:  map[string]*clientcmdapi.Cluster{"test": {Server: "https://test/clusters/root:foo"}},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{"test": {Token: "test"}},
+			},
+			existingWorkspaces: []string{"bar"},
+			newWorkspaceName:   "bar",
+			useAfterCreation:   true,
+			wantErr:            true,
+		},
+		{
+			name: "create, use after creation, ignore existing",
```

Comment:
I don't think this test case is testing what the description says? It's missing `existingWorkspaces: []string{"bar"}`.

Reply:
done